pkg/remote/client_lookup.go

```diff
@@ -25,6 +26,7 @@ func NewClientLookup(kcp Client, cache *ClientCache, strategy v1beta2.SyncStrate
 func (l *ClientLookup) Lookup(ctx context.Context, key client.ObjectKey) (Client, error) {
 	remoteClient := l.cache.Get(key)
 	if remoteClient != nil {
+		logf.FromContext(ctx).V(log.InfoLevel).Info("Client is not found in cache.")
```

Comment:
Perhaps dumping the key is necessary. Otherwise, when there are many Kymas, you'll not be able to match this log message with a particular Kyma.